# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../spqmail/0001-Makefile-fixed-to-use-n-option-with-head-tail.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 62cbfe9769bf73c9232d36e20877d02eb1901875 Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Fri, 21 Oct 2011 13:22:48 +0200
Subject: [PATCH] Makefile: fixed to use -n option with head/tail

---
 Makefile |  170 +++++++++++++++++++++++++++++++-------------------------------
 1 files changed, 85 insertions(+), 85 deletions(-)

diff --git a/Makefile b/Makefile
index d5a0587..ca54fad 100644
--- a/Makefile
+++ b/Makefile
@@ -23,8 +23,8 @@ compile alloc_re.c alloc.h byte.h
 auto-ccld.sh: \
 conf-cc conf-ld warn-auto.sh
 	( cat warn-auto.sh; \
-	echo CC=\'`head -1 conf-cc`\'; \
-	echo LD=\'`head -1 conf-ld`\' \
+	echo CC=\'`head -n 1 conf-cc`\'; \
+	echo LD=\'`head -n 1 conf-ld`\' \
 	) > auto-ccld.sh
 
 auto-gid: \
@@ -72,7 +72,7 @@ scan.h fmt.h
 auto_break.c: \
 auto-str conf-break
 	./auto-str auto_break \
-	"`head -1 conf-break`" > auto_break.c
+	"`head -n 1 conf-break`" > auto_break.c
 
 auto_break.o: \
 compile auto_break.c
@@ -80,7 +80,7 @@ compile auto_break.c
 
 auto_patrn.c: \
 auto-int8 conf-patrn
-	./auto-int8 auto_patrn `head -1 conf-patrn` > auto_patrn.c
+	./auto-int8 auto_patrn `head -n 1 conf-patrn` > auto_patrn.c
 
 auto_patrn.o: \
 compile auto_patrn.c
@@ -88,7 +88,7 @@ compile auto_patrn.c
 
 auto_qmail.c: \
 auto-str conf-qmail
-	./auto-str auto_qmail `head -1 conf-qmail` > auto_qmail.c
+	./auto-str auto_qmail `head -n 1 conf-qmail` > auto_qmail.c
 
 auto_qmail.o: \
 compile auto_qmail.c
@@ -96,7 +96,7 @@ compile auto_qmail.c
 
 auto_spawn.c: \
 auto-int conf-spawn
-	./auto-int auto_spawn `head -1 conf-spawn` > auto_spawn.c
+	./auto-int auto_spawn `head -n 1 conf-spawn` > auto_spawn.c
 
 auto_spawn.o: \
 compile auto_spawn.c
@@ -104,7 +104,7 @@ compile auto_spawn.c
 
 auto_split.c: \
 auto-int conf-split
-	./auto-int auto_split `head -1 conf-split` > auto_split.c
+	./auto-int auto_split `head -n 1 conf-split` > auto_split.c
 
 auto_split.o: \
 compile auto_split.c
@@ -112,16 +112,16 @@ compile auto_split.c
 
 auto_uids.c: \
 auto-uid auto-gid conf-users conf-groups
-	( ./auto-uid auto_uida `head -1 conf-users` \
-	&&./auto-uid auto_uidd `head -2 conf-users | tail -1` \
-	&&./auto-uid auto_uidl `head -3 conf-users | tail -1` \
-	&&./auto-uid auto_uido `head -4 conf-users | tail -1` \
-	&&./auto-uid auto_uidp `head -5 conf-users | tail -1` \
-	&&./auto-uid auto_uidq `head -6 conf-users | tail -1` \
-	&&./auto-uid auto_uidr `head -7 conf-users | tail -1` \
-	&&./auto-uid auto_uids `head -8 conf-users | tail -1` \
-	&&./auto-gid auto_gidq `head -1 conf-groups` \
-	&&./auto-gid auto_gidn `head -2 conf-groups | tail -1` \
+	( ./auto-uid auto_uida `head -n 1 conf-users` \
+	&&./auto-uid auto_uidd `head -n 2 conf-users | tail -n 1` \
+	&&./auto-uid auto_uidl `head -n 3 conf-users | tail -n 1` \
+	&&./auto-uid auto_uido `head -n 4 conf-users | tail -n 1` \
+	&&./auto-uid auto_uidp `head -n 5 conf-users | tail -n 1` \
+	&&./auto-uid auto_uidq `head -n 6 conf-users | tail -n 1` \
+	&&./auto-uid auto_uidr `head -n 7 conf-users | tail -n 1` \
+	&&./auto-uid auto_uids `head -n 8 conf-users | tail -n 1` \
+	&&./auto-gid auto_gidq `head -n 1 conf-groups` \
+	&&./auto-gid auto_gidn `head -n 2 conf-groups | tail -n 1` \
 	) > auto_uids.c.tmp && mv auto_uids.c.tmp auto_uids.c
 
 auto_uids.o: \
@@ -130,7 +130,7 @@ compile auto_uids.c
 
 auto_usera.c: \
 auto-str conf-users
-	./auto-str auto_usera `head -1 conf-users` > auto_usera.c
+	./auto-str auto_usera `head -n 1 conf-users` > auto_usera.c
 
 auto_usera.o: \
 compile auto_usera.c
@@ -139,42 +139,42 @@ compile auto_usera.c
 binm1: \
 binm1.sh conf-qmail
 	cat binm1.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm1
 	chmod 755 binm1
 
 binm1+df: \
 binm1+df.sh conf-qmail
 	cat binm1+df.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm1+df
 	chmod 755 binm1+df
 
 binm2: \
 binm2.sh conf-qmail
 	cat binm2.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm2
 	chmod 755 binm2
 
 binm2+df: \
 binm2+df.sh conf-qmail
 	cat binm2+df.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm2+df
 	chmod 755 binm2+df
 
 binm3: \
 binm3.sh conf-qmail
 	cat binm3.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm3
 	chmod 755 binm3
 
 binm3+df: \
 binm3+df.sh conf-qmail
 	cat binm3+df.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> binm3+df
 	chmod 755 binm3+df
 
@@ -337,18 +337,18 @@ wait.h seek.h qmail.h substdio.h strerr.h substdio.h fmt.h
 config: \
 warn-auto.sh config.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh config.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> config
 	chmod 755 config
 
 config-fast: \
 warn-auto.sh config-fast.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh config-fast.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> config-fast
 	chmod 755 config-fast
 
@@ -368,9 +368,9 @@ compile date822fmt.c datetime.h fmt.h date822fmt.h
 datemail: \
 warn-auto.sh datemail.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh datemail.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> datemail
 	chmod 755 datemail
 
@@ -480,17 +480,17 @@ dot-qmail.5
 dot-qmail.5: \
 dot-qmail.9 conf-break conf-spawn
 	cat dot-qmail.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> dot-qmail.5
 
 elq: \
 warn-auto.sh elq.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh elq.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> elq
 	chmod 755 elq
 
@@ -710,14 +710,14 @@ compile hier.c auto_qmail.h auto_split.h auto_uids.h fmt.h fifo.h
 home: \
 home.sh conf-qmail
 	cat home.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> home
 	chmod 755 home
 
 home+df: \
 home+df.sh conf-qmail
 	cat home+df.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> home+df
 	chmod 755 home+df
 
@@ -895,9 +895,9 @@ readwrite.h open.h headerbody.h maildir.h strerr.h
 mailsubj: \
 warn-auto.sh mailsubj.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh mailsubj.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> mailsubj
 	chmod 755 mailsubj
 
@@ -997,9 +997,9 @@ compile open_write.c open.h
 pinq: \
 warn-auto.sh pinq.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh pinq.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> pinq
 	chmod 755 pinq
 
@@ -1037,14 +1037,14 @@ gen_alloc.h
 proc: \
 proc.sh conf-qmail
 	cat proc.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> proc
 	chmod 755 proc
 
 proc+df: \
 proc+df.sh conf-qmail
 	cat proc+df.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
 	> proc+df
 	chmod 755 proc+df
 
@@ -1055,9 +1055,9 @@ compile prot.c hasshsgr.h prot.h
 qail: \
 warn-auto.sh qail.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh qail.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> qail
 	chmod 755 qail
 
@@ -1104,9 +1104,9 @@ qmail-control.5
 qmail-control.5: \
 qmail-control.9 conf-break conf-spawn
 	cat qmail-control.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-control.5
 
 qmail-getpw: \
@@ -1122,9 +1122,9 @@ qmail-getpw.8
 qmail-getpw.8: \
 qmail-getpw.9 conf-break conf-spawn
 	cat qmail-getpw.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-getpw.8
 
 qmail-getpw.o: \
@@ -1167,9 +1167,9 @@ qmail-limits.7
 qmail-limits.7: \
 qmail-limits.9 conf-break conf-spawn
 	cat qmail-limits.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-limits.7
 
 qmail-local: \
@@ -1232,9 +1232,9 @@ qmail-newmrh.8
 qmail-newmrh.8: \
 qmail-newmrh.9 conf-break conf-spawn
 	cat qmail-newmrh.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-newmrh.8
 
 qmail-newmrh.o: \
@@ -1257,9 +1257,9 @@ qmail-newu.8
 qmail-newu.8: \
 qmail-newu.9 conf-break conf-spawn
 	cat qmail-newu.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-newu.8
 
 qmail-newu.o: \
@@ -1283,9 +1283,9 @@ qmail-pw2u.8
 qmail-pw2u.8: \
 qmail-pw2u.9 conf-break conf-spawn
 	cat qmail-pw2u.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-pw2u.8
 
 qmail-pw2u.o: \
@@ -1374,9 +1374,9 @@ exit.h
 qmail-qstat: \
 warn-auto.sh qmail-qstat.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh qmail-qstat.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> qmail-qstat
 	chmod 755 qmail-qstat
 
@@ -1468,9 +1468,9 @@ qmail-send.8
 qmail-send.8: \
 qmail-send.9 conf-break conf-spawn
 	cat qmail-send.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-send.8
 
 qmail-send.o: \
@@ -1544,9 +1544,9 @@ qmail-start.8
 qmail-start.8: \
 qmail-start.9 conf-break conf-spawn
 	cat qmail-start.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-start.8
 
 qmail-start.o: \
@@ -1586,9 +1586,9 @@ fmt.h ip.h lock.h error.h exit.h datetime.h now.h datetime.h
 qmail-upq: \
 warn-auto.sh qmail-upq.sh conf-qmail conf-break conf-split
 	cat warn-auto.sh qmail-upq.sh \
-	| sed s}QMAIL}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPLIT}"`head -1 conf-split`"}g \
+	| sed s}QMAIL}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPLIT}"`head -n 1 conf-split`"}g \
 	> qmail-upq
 	chmod 755 qmail-upq
 
@@ -1599,9 +1599,9 @@ qmail-users.5
 qmail-users.5: \
 qmail-users.9 conf-break conf-spawn
 	cat qmail-users.9 \
-	| sed s}QMAILHOME}"`head -1 conf-qmail`"}g \
-	| sed s}BREAK}"`head -1 conf-break`"}g \
-	| sed s}SPAWN}"`head -1 conf-spawn`"}g \
+	| sed s}QMAILHOME}"`head -n 1 conf-qmail`"}g \
+	| sed s}BREAK}"`head -n 1 conf-break`"}g \
+	| sed s}SPAWN}"`head -n 1 conf-spawn`"}g \
 	> qmail-users.5
 
 qmail.0: \
-- 
1.6.6.2

